test/cql-pytest/util.py

```diff
@@ -295,8 +295,18 @@ def __init__(self, cql, key, value):
         self._original_value = None
 
     def __enter__(self):
-        self._original_value = literal_eval(
-            self._cql.execute(f"SELECT value FROM system.config WHERE name='{self._key}'").one().value)
+        res = self._cql.execute(f"SELECT type, value FROM system.config WHERE name='{self._key}'").one()
+        type = res.type
+        value = res.value
+        if type == 'bool':
+            if value.lower() in ['true', '1', 'yes', 'on']:
+                self._original_value = True
+            elif value.lower() in ['false', '0', 'no', 'off']:
+                self._original_value = False
+            else:
+                raise ValueError(f"Invalid boolean value '{value}'")
+        else:
```

Comment:
Why is all these needed? Wouldn't saving the original string work?

Reply:
no, the current code calls `boost::lexical_cast<bool>` to save the cooked value, but it is very minimalistic and expects `0` or `1`.